src/btree/bt_page.c

```diff
@@ -779,7 +773,18 @@ __page_inmem_prepare_update(WT_SESSION_IMPL *session, WT_ITEM *value, WT_CELL_UN
      * to ensure that written prepared delete operation must be removed from the data store, when
      * the prepared transaction gets rollback.
      */
-    if (WT_TIME_WINDOW_HAS_STOP(&unpack->tw)) {
+    if (WT_TIME_WINDOW_HAS_START_PREPARE(&unpack->tw)) {
+        upd->prepared_id = unpack->tw.start_prepared_id;
+        upd->prepare_ts = unpack->tw.start_prepare_ts;
+        upd->durable_ts = WT_TS_NONE;
```

Comment:
This is set twice. Can we ensure we only set this once? May be move the above setting to the else code?